pkg/datastore/datastore.go

```diff
@@ -166,13 +207,49 @@ func (ds *memoryDatastore) RemoveOperatorSource(uid types.UID) {
 	delete(ds.rows, uid)
 }
 
-func (ds *memoryDatastore) GetOperatorSource(opsrcUID types.UID) (opsrc *v1alpha1.OperatorSourceSpec, ok bool) {
+func (ds *memoryDatastore) GetOperatorSource(opsrcUID types.UID) (*OperatorSourceKey, bool) {
 	row, exists := ds.rows[opsrcUID]
 	if !exists {
 		return nil, false
 	}
 
-	return row.Spec, true
+	return &row.OperatorSourceKey, true
+}
+
+func (ds *memoryDatastore) OperatorSourceHasUpdate(opsrcUID types.UID, metadata []*RegistryMetadata) (bool, error) {
+	// TODO: Return fine grained information that describes repository that
+	// was removed, added or has a new release.
+	row, exists := ds.rows[opsrcUID]
+	if !exists {
+		return false, fmt.Errorf("datastore has no record of the specified OperatorSource [%s]", opsrcUID)
+	}
+
+	// Right now we consider remote and local operator source to be same only
```

Comment:
Not sure how useful this is. It is a repeat of what you have in the function description: `following event(s) indicate that a remote registry has been updated.`, whereas here you have it as when a remote registry has **not** been updated

Reply:
I removed it.